client/engine/chainservice/chainservice.go

```diff
@@ -73,7 +73,7 @@ type ChainService interface {
 	// SubscribeToEvents creates and returs a subscription channel.
 	SubscribeToEvents(types.Address) <-chan Event
 	// SendTransaction is for sending transactions with the chain service
-	SendTransaction(protocols.ChainTransaction)
```

Comment:
The `EthChainService` implementation of this method was previously returning a `[]*ethTypes.Transaction`. Why is this not a type error? Are we allowed to return whatever we like if nothing is specified here?

Reply:
At the moment, only `SimulatedBackendChainService` and `MockChainService` are used by the tests and the clients. So `EthChainService` was never used where a `ChainService` interface is required.

`EthChainService` will be used in production environments. So this PR modifies `EthChainService` to adhere to the `ChainService` interface.